src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -788,6 +788,24 @@ private bool IsValidExtensionMethodSignature()
                 this.ParameterRefKinds.IsDefault && // No 'ref' or 'out'
                 !this.IsParams();
 
+        private bool IsValidUserDefinedOperatorIs()
+        {
+            foreach (var parameter in this.Parameters)
+            {
+                if (parameter.RefKind != ((parameter.Ordinal == 0) ? RefKind.None : RefKind.Out))
+                {
+                    return false;
+                }
+            }
+
+            return
+                (this.ReturnsVoid || this.ReturnType.SpecialType != SpecialType.System_Boolean) &&
+                !this.IsGenericMethod &&
+                !this.IsVararg &&
+                this.ParameterCount > 0 &&
```

Comment:
That behavior inconsistency around existing user-defined operators is likely caused by legacy behavior, which we don't have for the ```is``` operator and I cannot think of any good reason to be inconsistent for it. I strongly believe that requirements between PE and source symbols should be identical. If you want to duplicate the logic, fine, but make it identical in this change. Once this is done, feel free to bring proposal about relaxing rules for PE symbols to design meeting.